components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/util/EventHookHandlerUtils.java

```diff
@@ -0,0 +1,265 @@
+        }
+
+        SecurityEventTokenPayload securityEventTokenPayload = new SecurityEventTokenPayload();
+        securityEventTokenPayload.setIss(getReference("", ""));
+        securityEventTokenPayload.setIat(System.currentTimeMillis());
+        securityEventTokenPayload.setJti(UUID.randomUUID().toString());
+        securityEventTokenPayload.setRci(getCorrelationID());
+        Map<String, EventPayload> eventMap = new HashMap<>();
+        eventMap.put(eventUri, eventPayload);
+        securityEventTokenPayload.setEvent(eventMap);
+        return securityEventTokenPayload;
+    }
+
+    /**
+     * Get correlation id from the MDC.
+     * If not then generate a random UUID, add it to MDC and return the UUID.
+     *
+     * @return Correlation id
+     */
+    public static String getCorrelationID() {
+
+        String correlationID = MDC.get(CORRELATION_ID_MDC);
+        if (StringUtils.isBlank(correlationID)) {
+            correlationID = UUID.randomUUID().toString();
+            MDC.put(CORRELATION_ID_MDC, correlationID);
+        }
+        return correlationID;
+    }
+
+    private static void setLocalUserClaimsToAuthenticatedUser(AuthenticatedUser authenticatedUser,
+                                                       AuthenticationContext context) {
+
+        Map<String, String> claimMappings = (Map<String, String>) context.getParameters()
+                    .get(Constants.SP_TO_CARBON_CLAIM_MAPPING);
+
+            if (claimMappings == null) {
+                log.debug("No local claim mappings found for the authenticated user from the context.");
+                return;
+            }
+
+            Map<ClaimMapping, String> userAttributes = authenticatedUser.getUserAttributes();
+
+            if (userAttributes == null) {
+                userAttributes = new HashMap<>();
+            }
+
+            for (Map.Entry<ClaimMapping, String> entry : userAttributes.entrySet()) {
+
+                ClaimMapping claimMapping = entry.getKey();
+                Claim localClaim = claimMapping.getLocalClaim();
+
+                if (claimMappings.containsKey(localClaim.getClaimUri())) {
+                    localClaim.setClaimUri(claimMappings.get(localClaim.getClaimUri()));
+                }
+            }
+    }
+
+    /**
+     * Retrieve the reference for the given endpoint and id.
+     *
+     * @param endpoint     Endpoint.
+     * @param id           Resource id.
+     * @return Resource location.
+     */
+    public static String getReference(String endpoint, String id) {
+
+        return getURL(endpoint) + "/" + id;
+    }
+
+    private static String getURL(String endpoint) {
+
+        try {
+            return ServiceURLBuilder.create().addPath(endpoint).build().getAbsolutePublicURL();
+        } catch (URLBuilderException e) {
+            log.debug("Error occurred while building the endpoint URL with tenant " +
+                        "qualified URL.", e);
+        }
+        return endpoint;
+    }
```

Comment:
what's the out come of this function ? what would be the endpoint passed

Reply:
to get the tenanted url. If any exception occurred while building the URL then return the passed argument.